src/sustainable-web-design.js

```diff
@@ -52,14 +48,17 @@ class SustainableWebDesign {
    * Accept an object keys by the different system components, and
    * return an object with the co2 figures key by the each component
    *
-   * @param {object} energyByComponent - energy grouped by the four system components
-   * @param {number} [carbonIntensity] - carbon intensity to apply to the datacentre values
-   * @return {number} the total number in grams of CO2 equivalent emissions
+   * @template {Record<string, number>} EnergyObject
+   * @param {EnergyObject} energyByComponent - energy grouped by the four system components
+   * // TODO (simon) check on this type for carbonIntensity
+   * @param {(number | boolean)=} carbonIntensity - carbon intensity to apply to the datacentre values
```

Comment:
question: it looks like this param is being used as a boolean but is initialized to a number when not provided. i'm wondering if it should be initialized to `false` instead?